packages/contracts/contracts/zkopru/controllers/UserInteractable.sol

```diff
@@ -178,17 +181,19 @@ contract UserInteractable is Storage {
         uint256[3] memory resultHashInputs = [spendingPubKey, salt, assetHash];
         uint256 note = Poseidon3.poseidon(resultHashInputs);
         // Receive token
-        if (token != address(0) && amount != 0) {
-            try
-                IERC20(token).transferFrom(msg.sender, address(this), amount)
-            {} catch {
-                revert("Transfer ERC20 failed");
-            }
-        } else if (token != address(0)) {
-            try
-                IERC721(token).transferFrom(msg.sender, address(this), nft)
-            {} catch {
-                revert("Transfer NFT failed");
+        if (token != address(0)) {
+            if (chain.registeredERC20s[token]) {
+                require(nft == 0, "ERC20 note cannot have NFT");
+                IERC20(token).safeTransferFrom(
```

Comment:
The openzeppelin [ERC20 interface](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/IERC20.sol) doesn't include a `safeTransferFrom` function.

Reply:
This call will not revert if it fails, it will silently return false. A user could falsely deposit infinite tokens without a revert check.

Reply:
Line 19 `using SafeERC20 for ERC20` is adding the `safeTransferFrom` method to IERC20 interface